lib/web_ui/lib/src/engine/platform_views/content_manager.dart

```diff
@@ -127,7 +127,34 @@ class PlatformViewManager {
   /// never been rendered before.
   void clearPlatformView(int viewId) {
     // Remove from our cache, and then from the DOM...
-    _contents.remove(viewId)?.remove();
+    final html.Element? element = _contents.remove(viewId);
+    _safelyRemoveSlottedElement(element);
+  }
+
+  // We need to remove slotted elements like this because of a Safari bug that
+  // gets triggered when a slotted element is removed in a JS tick different
+  // than its slot (after the slot is removed).
+  //
+  // TODO(web): Cleanup https://github.com/flutter/flutter/issues/85816
+  void _safelyRemoveSlottedElement(html.Element? element) {
+    if (element == null) {
+      return;
+    }
+    if (browserEngine != BrowserEngine.webkit) {
+      element.remove();
+      return;
+    }
+    final String slotName = "tombstone-${element.getAttribute('slot')}";
+    // Create and inject a new slot in the Shadow Root
```

Comment:
nit: no need to capitalize "shadow root" (it's just a technical term, like class, function, widget, element, or render object).